deploy/lib/xquery/setup.xqy

```diff
@@ -2391,18 +2330,27 @@ declare function setup:add-range-field-indexes-helper(
   $db-config as element(db:database)) as element(configuration)
 {
   if ($db-config/db:range-field-indexes/db:range-field-index) then
-    xdmp:eval('
-      import module namespace admin = "http://marklogic.com/xdmp/admin" at "/MarkLogic/admin.xqy";
-      declare namespace db="http://marklogic.com/xdmp/database";
-      declare variable $admin-config external;
-      declare variable $database external;
-      declare variable $db-config external;
-      admin:database-add-range-field-index($admin-config, $database, $db-config/db:range-field-indexes/db:range-field-index)',
-      (
-        xs:QName("admin-config"), $admin-config,
-        xs:QName("database"), $database,
-        xs:QName("db-config"), $db-config
-      ))
+    admin:database-add-range-field-index(
+      $admin-config,
+      $database,
+      for $index in $db-config/db:range-field-indexes/db:range-field-index
+      return
+        if (setup:at-least-version("6.0-0")) then
+          admin:database-range-field-index(
+            $index/db:scalar-type,
+            $index/db:field-name,
+            ($index/db:collation/fn:string(), "")[1], (: ML6 requires xs:string; later requires xs:string? :)
+            $index/db:range-value-positions,
+            $index/db:invalid-values
+          )
+        else
+          admin:database-range-field-index(
+            $index/db:scalar-type,
+            $index/db:field-name,
+            $index/db:collation,
+            $index/db:range-value-positions
```

Comment:
@dmcassel Wow, I totally misread that one. Deleted a lot of my other comments. Thanks.